recipe/meta.yaml

```diff
@@ -13,43 +14,243 @@ source:
 build:
   number: 0
   noarch: python
-  script: {{ PYTHON }} -m pip install . --no-deps -vv
-  entry_points:
-    - parsl-globus-auth = parsl.data_provider.globus:cli_run
 
 requirements:
   host:
-    - python >=3.5
+    - python >=3.6
     - pip
   run:
-    - python >=3.5
-    - pyzmq >=17.1.2
-    - typeguard >=2.5
-    - ipyparallel
-    - globus-sdk
-    - dill
-    - tblib
-    - ipykernel
-    - requests
-    - paramiko
-    - psutil >=5.5.1
-    - sqlalchemy >=1.3.0,!=1.3.4,<1.4
-    - sqlalchemy-utils
-    - pydot
-    - networkx
-    - flask >=1.0.2
-    - flask-sqlalchemy
-    - pandas
-    - plotly
-    - python-daemon
-
-test:
-  requires:
-    - pip
-  imports:
-    - parsl
-  commands:
-    - pip check
+    - python >=3.6
+
+outputs:
+  - name: {{ name }}
+    script: install_parsl.sh
+    build:
+      noarch: python
+      entry_points:
+        - parsl-globus-auth = parsl.data_provider.globus:cli_run
+
+    requirements:
+      host:
+        - python >=3.6
+        - pip
+      run:
+        - python >=3.6
+        # requirements.txt
+        - pyzmq >=17.1.2
+        - typeguard >=2.10
+        - typing-extensions
+        - globus-sdk
+        - dill
+        - tblib
+        - requests
+        - paramiko
+        - psutil >=5.5.1
+
+    test:
+      requires:
+        - pip
+      imports:
+        - parsl
+      commands:
+        - pip check
+
+  - name: {{ name }}-with-monitoring
+    build:
+      noarch: python
+    requirements:
+      host:
+        - python >=3.6
+      run:
+        - python >=3.6
+        - {{ pin_subpackage(name, max_pin='x.x.x.x.x.x') }}
+        - sqlalchemy >=1.3.0,!=1.3.4,<1.4
+        - sqlalchemy-utils
+        - pydot
+        - networkx
+        - flask >=1.0.2
+        - flask-sqlalchemy
+        - pandas
+        - plotly
+        - python-daemon
+    test:
+      imports:
+        - parsl
+        - parsl.monitoring
+      commands:
+        - pip check
+      requires:
+        - pip
+
+  - name: {{ name }}-with-aws
+    build:
+      noarch: python
+    requirements:
+      host:
+        - python >=3.6
+      run:
+        - python >=3.6
+        - {{ pin_subpackage(name, max_pin='x.x.x.x.x.x') }}
+        - boto3
+    test:
+      imports:
+        - parsl
+      commands:
+        - pip check
+      requires:
+        - pip
+
+  - name: {{ name }}-with-kubernetes
+    build:
+      noarch: python
+    requirements:
+      host:
+        - python >=3.6
+      run:
+        - python >=3.6
+        - {{ pin_subpackage(name, max_pin='x.x.x.x.x.x') }}
+        - kubernetes
+    test:
+      imports:
+        - parsl
+      commands:
+        - pip check
+      requires:
+        - pip
+
+# oauth-ssh not on conda-forge
+#  - name: {{ name }}-with-oauth_ssh
+#    build:
+#      noarch: python
+#    requirements:
+#      host:
+#        - python >=3.6
+#      run:
+#        - python >=3.6
+#        - {{ pin_subpackage(name, max_pin='x.x.x.x.x.x') }}
+#        - oauth-ssh >=0.9
```

Comment:
`oauth-ssh` doesn't appear to be available on conda-forge.  Is there an equivalent package that we should be using here?

Reply:
I'd skip this optional extra completely. There isn't an alternative for this package because this addresses a niche issue in the research computing space. I'll make a note of this in our docs.